src/mlpack/methods/ann/init_rules/variance_scaling_init.hpp

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file variance_scaling_init.hpp
+ * @author Shashank Shekhar
+ *
+ * Intialization rule for the neural networks.
+ * VarianceScalingNormalInit is performed by assigning a gaussian matrix with
+ * zero mean and variance given by (scaling factor / n) to the weight matrix.
+ * VarianceScalingUniformInit is performed by assigning a random matrix between
+ * [-limit, limit] where limit is given as sqrt(3*scaling facor/n).
+ * n is decided by the mode of initializer and is described below.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_VARIANCE_SCALING_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_VARIANCE_SCALING_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include "gaussian_init.hpp"
+#include "random_init.hpp"
+
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+class FanIn
```

Comment:
Can you clarify for what each class is used for?

Reply:
I had described the classes later on but I think it makes sense to remove the two descriptions below and add a single one here.